src/main/java/coresearch/cvurl/io/request/CVurlProxyType.java

```diff
@@ -0,0 +1,20 @@
+package coresearch.cvurl.io.request;
+
+import java.net.Proxy;
+
+public enum CVurlProxyType {
```

Comment:
Because constructor of `java.net.Proxy` requires an instance of `java.net.SocketAddress`. My intention was to hide it's creation. It looks a bit cleaner with a wrapper:

        final Response<String> firstResponse = cvurl.get("https://google.com/")
                .withProxy(CVurlProxy.of(CVurlProxyType.HTTP, "proxy.host", 8080))
                .asString()
                .orElse(null);
        final Response<String> secondResponse = cvurl.get("https://google.com/")
                .withProxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("proxy.host", 8080)))
                .asString()
                .orElse(null);

Reply:
I am talking about using Proxy.Type instead of CVurlProxyType inside of CVurlProxy

Reply:
removed the `CVurlProxyType` enum and replaced uses directly with `Proxy.Type`